File: java/lambda-cron/src/main/java/software/amazon/awscdk/examples/LambdaCronStack.java
package software.amazon.awscdk.examples;

import java.util.UUID;
import software.constructs.Construct;
import software.amazon.awscdk.Duration;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.services.events.Rule;
import software.amazon.awscdk.services.events.Schedule;
import software.amazon.awscdk.services.events.targets.LambdaFunction;
import software.amazon.awscdk.services.lambda.Code;
import software.amazon.awscdk.services.lambda.Runtime;
import software.amazon.awscdk.services.lambda.SingletonFunction;

/** Lambda Cron CDK example for Java! */
class LambdaCronStack extends Stack {
  public LambdaCronStack(final Construct parent, final String name) {
    super(parent, name);

    SingletonFunction lambdaFunction =
        SingletonFunction.Builder.create(this, "cdk-lambda-cron")
            .description("Lambda which prints \"I'm running\"")
            .code(Code.fromInline("def main(event, context):\n" + "    print(\"I'm running!\")\n"))
            .handler("index.main")
            .timeout(Duration.seconds(300))
            .runtime(Runtime.PYTHON_3_9)
            .uuid(UUID.randomUUID().toString())
            .build();

    Rule rule =
        Rule.Builder.create(this, "cdk-lambda-cron-rule")
            .description("Run every day at 6PM UTC")
            .schedule(Schedule.expression("cron(0 18 ? * MON-FRI *)"))
            .build();

    rule.addTarget(new LambdaFunction(lambdaFunction));
  }
}

File: csharp/lambda-cron/src/LambdaCron/LambdaCronStack.cs
using Amazon.CDK;
using Amazon.CDK.AWS.Events;
using Amazon.CDK.AWS.Events.Targets;
using Amazon.CDK.AWS.Lambda;
using System.IO;
using System.Text;

namespace LambdaCron
{
  public class LambdaCronStack : Stack
  {
    internal LambdaCronStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
    {
      string currentDirectory = Directory.GetCurrentDirectory();
      string filePath = Path.Combine(currentDirectory, "src/LambdaCron/lambda-handler.py");
      string inlineCodeFromFile = File.ReadAllText(filePath, Encoding.UTF8);

      var lambdaFunction = new Function(this, "Singleton", new FunctionProps()
      {
        Code = new InlineCode(inlineCodeFromFile),
        Handler = "index.main",
        Timeout = Duration.Seconds(300),
        Runtime = Runtime.PYTHON_3_6
      });

      // Run 6:00 PM UTC every Monday through Friday
      // See https://docs.aws.amazon.com/lambda/latest/dg/tutorial-scheduled-events-schedule-expressions.html
      var rule = new Rule(this, "Rule", new RuleProps()
      {
        Schedule = Schedule.Expression("cron(0 18 ? * MON-FRI *)")
      });

      rule.AddTarget(new LambdaFunction(lambdaFunction));
    }
  }
}
